core/src/worker/client.rs

```diff
@@ -48,11 +48,39 @@ impl WorkerClientBag {
             use_versioning,
         }
     }
-    fn versioning_build_id(&self) -> String {
-        if self.use_versioning {
+
+    fn default_capabilities(&self) -> Capabilities {
```

Comment:
nit: Why don't you just initialize it as empty and overwrite when system info resolves instead of dealing with optionals?

Reply:
There are other places which like to know if it was ever set by reading from server